### DISCUSSION TITLE
Documentation for all packages

### DISCUSSION THREAD

**Discussion by nimit2801** [octokit/octokit.js#2135] [Q&A] [answered]
Where can I find documentation for all projects associated with octokit.js, core, rest, webhook, etc

Thank You

**Comment by wolfy1339** [comment]
There isn't one spot for all of the documentation. It's across all the different modules' repo
The only exception, is `@octokit/rest` which it's documentation lies here: https://octokit.github.io/rest.js/v18

**Comment by gr2m** [comment, marked as answer]
Sorry for the lack of a complete documentation, we didn't get to it yet. 

A good starting point is https://github.com/octokit/octokit.js/, it also links to the underlying repositories and how they are combined together into the `octokit` package, so you e.g. if you are interested in building a GitHub App, you can look into https://github.com/octokit/app.js

Do you have any specific question?

**Reply by nimit2801** [reply]
Was looking for implementing the wrappers and webhooks for a project from GitHub.
To be honest these are very cool wrappers and would love to see some mainstream documentation. Also as beginners, if someone is looking for the wrappers they won't be able to use the tool properly.

 Would love to contribute to the documentation of the project 💖

**Reply by gr2m** [reply]
How about you create a blog post about what you want to build and I help you fill in the gaps, and we can use that as an artifact for a future documentation guide? Or we create a folder `guides` in this repository to start gathering them all in this single place?

**Reply by nimit2801** [reply]
That is an awesome 💯 idea, I'll start implementing the wrapper with my project and would write an article for the same. I'm not used to the whole wrappers yet so would use this thread to ask questions and update about the 📖 blog post.

**Reply by nimit2801** [reply]
```js
// install with: npm install @octokit/webhooks
import { Webhooks, createNodeMiddleware} from "@octokit/webhooks";
const webhooks = new Webhooks({
  secret: "xoxo",
});

webhooks.onAny(({ id, name, payload }) => {
  console.log(name, "event received");
});
webhooks.on("issues", ({id, name, payload}) => {
    console.log(id, name, payload, "received");
});

import http from "http";
http.createServer(createNodeMiddleware(webhooks)).listen(3000, () => {
    console.log('running on http://localhost:3000');
});
// can now receive webhook events at /api/github/webhooks
```

I'm not receiving anything in my terminal

**Reply by wolfy1339** [reply]
The secret is not a Personal Access Token, but a password that you set for the webhooks themselves in the config on GitHub

**Reply by nimit2801** [reply]
It's giving me 404 on GitHub when I try to send one
![image](https://user-images.githubusercontent.com/53462249/126407133-66f0faaa-6f82-498f-8ac4-53506ac5ab2c.png)

**Reply by wolfy1339** [reply]
Can you use the server's URL/IP directly?

Did you set the URL to include the /api/github/webhooks path

**Reply by nimit2801** [reply]
Directly it says cannot get which is obvious because we're not sending any response on `'/'`. I've exactly tried the example in the documentation 
![image](https://user-images.githubusercontent.com/53462249/126407469-7813f7d1-6944-42cf-a10f-b9c6c841be3b.png)
I thought this would work directly.
I'm giving it a try with express and manually handling things with `post method` on `/payload` method.

**Reply by wolfy1339** [reply]
What I mean with directly is not using ngrok, and simply use the server's IP with the path

**Reply by gr2m** [reply]
It looks like you like you set the webhook URL to the ngrok URL

<img width="332" alt="image" src="https://user-images.githubusercontent.com/39992/126409865-e0725149-a422-4ff4-b953-2f00a1168197.png">

But you need to set it to `https://[ngrok id].ngrok.io/api/github/webhooks`. That's what the last comment in the usage example means.

**Reply by nimit2801** [reply]
~~And do I've to add `/payload` at the end?~~
Got it we don't have to. Thanks, It's working!
![image](https://user-images.githubusercontent.com/53462249/126429351-292031a9-0bf0-468e-84fe-d485be8f689f.png)

Also saw your video on: https://youtu.be/H1TupRbbjSI was amazing!

**Comment by tranhuusang9899** [comment]
Join SangKaa on Slack

Just a quick reminder that your invitation from ST 79 ***@***.***​com) is waiting. Don’t miss out — come join the conversation in the SangKaa workspace!

Join Now [https://join.slack.com/t/sangkaa/invite/enQtMjMxOTc0MjQ1NDI2MC05YWQwZDI5OTU2NDYzNDFhZDMyYjI5YzlhOWRiOGY3ZjBkNDVmMTljYjI3NmZjMzZhZDUyNzVjYTVhZDQ3NTk0?x=x-p2336928285456-2313236162482-2319642435011]
What is Slack? [https://slack.com/features]

Slack connects you with the people and apps you work with every day. Message with your team, share files, and get updates from other tools, all within Slack. Learn more [https://slack.com/features].

https://slack.com/features

Workspace name: SangKaa
Workspace URL: sangkaa.slack.com
